Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -418,6 +418,79 @@ protected NetworkIdentity GetSyncVarNetworkIdentity(uint netId, ref NetworkIdent
             return identityField;
         }
 
+        // helper function for [SyncVar] NetworkIdentities.
+        protected void SetSyncVarNetworkBehaviour<T>(T newBehaviour, ref T behaviourField, ulong dirtyBit, ref SyncNetBehaviour syncField) where T : NetworkBehaviour
+        {
+            if (getSyncVarHookGuard(dirtyBit))
+                return;
+
+            uint newNetId = 0;
+            if (newBehaviour != null)
+            {
+                newNetId = newBehaviour.netId;
+                if (newNetId == 0)
+                {
+                    logger.LogWarning($"{nameof(SetSyncVarNetworkBehaviour)} NetworkIdentity " + newBehaviour + " has a zero netId. Maybe it is not spawned yet?");
+                }
+            }
+
+            // todo update this log for NetworkBehaviours
+            if (logger.LogEnabled()) logger.Log("SetSyncVarNetworkIdentity NetworkIdentity " + GetType().Name + " bit [" + dirtyBit + "] netIdField:" + syncField + "->" + newNetId);
+            SetDirtyBit(dirtyBit);
+
+            int compIndex = newBehaviour.ComponentIndex;
```

Comment:
please use full name instead of abbrevations: componentIndex is more obvious to non native english speakers